src/index.ts

```diff
@@ -39,62 +24,33 @@ export class ThemeResolverPlugin {
             this.pathRegex.push(new RegExp(`^${res.prefix}/`));
         });
         this.cache = {};
-        this.chosenResolver = {};
     }
 
     public apply(resolver: any) {
-        const target = resolver.ensureHook("module");
-
-        resolver.hooks.module.tapAsync(
-            "ThemeResolverPlugin",
-            (request: any, resolveContext: any, callback: () => void) => {
-                this.pathRegex.forEach((reg, x) => {
-                    if (request.request.match(reg)) {
-                        this.chosenResolver = Object.assign(ThemeResolverPlugin.defaultOptions, this.options[x]);
+        const target = resolver.ensureHook("resolved");
+
+        resolver.getHook("module")
+        .tapAsync("ThemeResolverPlugin", (request: any, resolveContext: any, callback: () => void) => {
+                let chosenResolver = this.getResolver(request);
```

Comment:
shoudn't this be a `const`? It's not replaced anywhere afterwards.

Reply:
yes this should be a const